backend/src/packages/sites/site.entity.ts

```diff
@@ -0,0 +1,45 @@
+import { type IEntity } from '~/libs/interfaces/interfaces.js';
+
+class SiteEntity implements Omit<IEntity, 'toNewObject'> {
+  private 'id': number;
+
+  private 'name': string;
+
+  private 'publishedUrl': string | null;
+
+  private constructor({
+    id,
+    name,
+    publishedUrl,
+  }: {
+    id: number;
+    name: string;
+    publishedUrl: string | null;
+  }) {
```

Comment:
Let's move it to type as it's duplicating in several places
`upd` looks like you've already declared such type